doc/reset_factory/ResetFactoryHLD.md

```diff
@@ -0,0 +1,210 @@
+1. It should support resetting configuration to factory default settings.
+
+2. It should clear logs, tech-support and reboot-cause files, users history files and home directories.
+
+3. It should reboot the system to load and apply the new configuration.
+
+4. It should stop sonic.target to ensure that files are not deleted while other daemons are utilizing them.
+
+5. it should allow users/application to create their own factory default configurations or
+extend the reset factory functionality.
+
+
+####  1.5.2. <a name='ConfigurationManagementRequirements'></a>Configuration and Management Requirements
+We will create a command line tool /usr/bin/reset-factory which should provide following functionality:
+
+1. default: reset configurations to factory default, clear all system logs and files.
+
+2. keep-basic: preserves basic configurations only after boot. Logs and files will be deleted users history files and home directories.
+
+3. keep-all-config: preserves all configurations after boot. Logs and files will be deleted.
+
+4. only-config: reset configurations to factory default. Logs and files will be preserved.
+
+    
+##  2 <a name='Design'></a>Design
+	
+###  2.1 <a name='High-LevelDesign'></a>High-Level Design
+![diagram](diagram.png)
+
+(flow description in the chapter below)
+
+"reset-factory" script will create new config_db.json if needed using "config-setup factory" command.<br/>
+In addition, it will clear logs, tech-support and reboot-cause files, users history files and home directories.<br/>
+
+Steps:
+
+1. Stops sonic.target: 
+
+systemctl stop sonic.target
+
+2. If not "keep-all-config":
+
+config-setup factory <None/keep-basic/only-config>
+
+3. If not ("only-config" or “keep-basic”):
+
+* Delete bash , vim and python history files under "/home".
+* Delete any non-dotfiles in the home directories.
+
+4. If not "only-config":
+* Delete tech-support files under "/var/dump/"
+* Delete all files under "/var/log"
+* Delete all files and symlinks under "/host/reboot-cause/"
+
+5. Print to /var/log/systemlog:
+New log file that won't be cleared in reset-factory.
+It will allow user to track operations such as "reset-factory".
+
+6. Reboot
+
+
+## 3 <a name='Functionality'></a>Functionality
+###  3.1 <a name='Reset-Factory'></a>Reset-Factory
+
+##### Default
+It will generate the default config_db.json file by running command:<br/>
+config-setup factory<br/>
+It will clear system logs and files.
+
+##### Keep-basic
+
+It will generate the default config_db.json and merge it with basic configurations from the current “config_db.json”.<br/>
+Command: config-setup factory keep-basic<br/>
+It will clear system logs and files.<br/>
+
+##### Keep-all-config
+It will keep the same configuration but will clear system logs and files.
+
+##### Only-config
+It will generate the default config_db.json file by running command:<br/>
+config-setup factory only-config<br/>
+It won't clear system logs and files.
```

Comment:
Please see my [comment](https://github.com/sonic-net/SONiC/pull/1231#discussion_r1088065516)
I believe this will cover it.
But I would like to know that exact flow and path of folder, Can you please share the relevant HLD/File ?